sdks/java/io/rabbitmq/src/main/java/org/apache/beam/sdk/io/rabbitmq/RabbitMqIO.java

```diff
@@ -53,276 +38,235 @@
-        .setQueueDeclare(false)
-        .setExchangeDeclare(false)
+        // should force the user to define this, as this queue won't already exist
+        .setReadParadigm(
+            ReadParadigm.ExistingQueue.of("beam-unconfigured-read-queue-" + UUID.randomUUID()))
+        // this policy is only appropriate for pipelines with at-least-once semantics
+        // and capable of handling potentially a large number of repeatedly-delivered messages
+        .setRecordIdPolicy(RecordIdPolicy.alwaysUnique())
+        // unbounded reads
         .setMaxReadTime(null)
         .setMaxNumRecords(Long.MAX_VALUE)
-        .setUseCorrelationId(false)
+        // processing-time only policy by default; often not a great choice
+        .setTimestampPolicyFactory(TimestampPolicyFactory.withProcessingTime())
+        // only reasonable for local development
+        .setConnectionHandlerProviderFn(
+            new ConnectionProviderFromUri("amqp://guest:guest@localhost:5672/"))
         .build();
   }
 
   public static Write write() {
     return new AutoValue_RabbitMqIO_Write.Builder()
-        .setExchangeDeclare(false)
-        .setQueueDeclare(false)
+        .setConnectionHandlerProviderFn(
+            new ConnectionProviderFromUri("amqp://guest:guest@localhost:5672/"))
         .build();
   }
 
   private RabbitMqIO() {}
 
-  private static class ConnectionHandler {
-
-    private final ConnectionFactory connectionFactory;
-    private Connection connection;
-    private Channel channel;
-
-    public ConnectionHandler(String uri)
-        throws URISyntaxException, NoSuchAlgorithmException, KeyManagementException {
-      connectionFactory = new ConnectionFactory();
-      connectionFactory.setUri(uri);
-      connectionFactory.setAutomaticRecoveryEnabled(true);
-      connectionFactory.setConnectionTimeout(60000);
-      connectionFactory.setNetworkRecoveryInterval(5000);
-      connectionFactory.setRequestedHeartbeat(60);
-      connectionFactory.setTopologyRecoveryEnabled(true);
-      connectionFactory.setRequestedChannelMax(0);
-      connectionFactory.setRequestedFrameMax(0);
-    }
-
-    public void start() throws TimeoutException, IOException {
-      connection = connectionFactory.newConnection();
-      channel = connection.createChannel();
-      if (channel == null) {
-        throw new IOException("No RabbitMQ channel available");
-      }
-    }
-
-    public Channel getChannel() {
-      return this.channel;
-    }
-
-    public void stop() throws IOException {
-      if (channel != null) {
-        try {
-          channel.close();
-        } catch (Exception e) {
-          // ignore
-        }
-      }
-      if (connection != null) {
-        connection.close();
-      }
-    }
-  }
-
   /** A {@link PTransform} to consume messages from RabbitMQ server. */
   @AutoValue
   public abstract static class Read extends PTransform<PBegin, PCollection<RabbitMqMessage>> {
-    @Nullable
```

Comment:
Instead of a bunch of disparate properties to aggregate together somehow later, I've broken these into a known set of `ReadParadigm`s.